socorro/external/postgresql/setupdb_app.py

```diff
@@ -233,9 +212,35 @@ def generate_fakedata(self, db, fakedata_days):
             """, dict(zip(["one", "two", "three", "four"],
                       list(fakedata.featured_versions))))
 
-    def main(self):
+    def construct_db_url(self, dbname=None, superuser=False):
+        """Takes a URL to connect to Postgres and updates database name
+            or superuser name/password as indicated"""
+        database_username = self.config.get('database_username')
+        database_password = self.config.get('database_password')
+        database_hostname = self.config.get('database_hostname')
+        database_port = self.config.get('database_port')
+        database_name = dbname
+
+        if superuser:
+            database_username = self.config.get('database_superusername')
+            database_password = self.config.get('database_superuserpassword')
+
+        # construct a URL
+        url = 'postgresql://'
+        if database_username:
+            url += '%s' % database_username
+        if database_password:
+            url += ':%s' % database_password
+        url += '@'
```

Comment:
I hadn't noticed that ```setupdb_app.py``` effectively has no tests.  In addition, the file that does much of the work, ```postgresqlalchemymanager.py```, doesn't even have a test file.